src/ottplan.java

```diff
@@ -0,0 +1,203 @@
+    } 
+    double getTotalAmount()
+    {
+        return subscriptionFee;
+    }
+    void setTotalAmount(double newAmount)
+    {
+        this.subscriptionFee=newAmount;
+    }
+    void setEligibleForDiscount(String eligible)
+    {
+        this.discountEligibilty=eligible;
+    }
+}
+class Prime extends Subscription{
+    @Override
+    public void calculate(){
+        this.planName=namesOfPlans.prime.getValue();
+        this.maximumConcurrentViewers=maxConView.prime.getValue();
+        this.resolution=supportedResolution.prime.getValue();
+        this.subscriptionFee=priceslab.prime.getValue();
+        this.subscriptionFee+=(taxslab.prime.getValue()*this.subscriptionFee);
+    }
+}
+class Premium extends Subscription{
+    @Override
+    public void calculate() {
+        this.planName=namesOfPlans.premium.toString();
+        this.maximumConcurrentViewers=maxConView.premium.getValue();
+        this.resolution=supportedResolution.premium.toString();
+        this.subscriptionFee=priceslab.premium.getValue();
+        this.subscriptionFee+=(taxslab.premium.getValue()*this.subscriptionFee);
+    }
+}
+class Platinum extends Subscription{
+    @Override
+    public void calculate() {
+        this.planName=namesOfPlans.platinum.getValue();
+        this.maximumConcurrentViewers=maxConView.platinum.getValue();
+        this.resolution=supportedResolution.platinum.getValue();
+        this.subscriptionFee=priceslab.platinum.getValue();
+        this.subscriptionFee+=(taxslab.platinum.getValue()*this.subscriptionFee);
+    }
+}
+class Factory{
+    static Map<String,Subscription> planMap=new HashMap<>();
+    static {
+        planMap.put("prime",new Prime());
+        planMap.put("premium",new Premium());
+        planMap.put("platinum",new Platinum());
+    }
+    public static Subscription getPlan(String operator)
+    {
+        return planMap.get(operator.toLowerCase());
+    }
+}
+class discountEligible{
+    boolean hasConsecutiveCharacters(String name)
+    {
+        for(int i=1;i<name.length();i++)
+        {
+            if(name.charAt(i)==name.charAt(i-1))
+                return true;
+        }
+        return false;
+    }
+    boolean hasTwoDistinctVowels(String name)
+    {
+        Map<Character,Integer> vowelMap=new HashMap<Character,Integer>();
+        for(int i=0;i<name.length();i++)
+        {
+            char ch=name.charAt(i);
+            if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u')
+            {
+                if(vowelMap.containsKey(ch)){
+                    vowelMap.put(ch,vowelMap.get(ch)+1);
+                }
+                else{
+                    vowelMap.put(ch,1);
+                }
+            }
+        }
+        if(vowelMap.size()>=2)
+        {
+            return true;
+        }
+        return false;
+    }
+    boolean isConsonant(char ch)
+    {
+        if(ch=='a'|| ch=='e'||ch=='i'||ch=='o'||ch=='u')
+            return false;
```

Comment:
Create a `isVowel(ch)` function. It's being used by another function also right?